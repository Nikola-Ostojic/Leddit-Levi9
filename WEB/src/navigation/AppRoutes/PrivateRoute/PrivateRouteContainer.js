import { connect } from "react-redux";

import { getUsername } from "modules/Auth/AuthSelectors";

import PrivateRoute from "./PrivateRoute";

export const mapStateToProps = (state) => ({
  user: getUsername(state)
});

export default connect(mapStateToProps)(PrivateRoute);
