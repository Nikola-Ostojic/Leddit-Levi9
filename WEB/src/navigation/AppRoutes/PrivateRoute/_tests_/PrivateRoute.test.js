import React from "react";
import { shallow } from "enzyme";

import { Route } from "react-router-dom";

import PrivateRoute from "../PrivateRoute";

const TestComponent = () => <div></div>;

describe("PrivateRoute test", () => {
  const mockedNavProps = { location: "/auth" };
  it("Should match snapshot when user does not exists", () => {
    const wrapper = shallow(<PrivateRoute component={TestComponent} />)
      .find(Route)
      .renderProp("render")(mockedNavProps);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should match snapshot when user does exists", () => {
    const wrapper = shallow(
      <PrivateRoute component={TestComponent} user="test" />
    )
      .find(Route)
      .renderProp("render")(mockedNavProps);
    expect(wrapper).toMatchSnapshot();
  });
});
