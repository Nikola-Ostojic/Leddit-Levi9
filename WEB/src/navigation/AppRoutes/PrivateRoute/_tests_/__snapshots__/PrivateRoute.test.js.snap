// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute test Should match snapshot when user does exists 1`] = `
<TestComponent
  location="/auth"
/>
`;

exports[`PrivateRoute test Should match snapshot when user does not exists 1`] = `
<Redirect
  to={
    Object {
      "pathname": "/auth",
      "state": Object {
        "from": "/auth",
      },
    }
  }
/>
`;
